providers/provider-springmvc/src/main/java/io/servicecomb/provider/springmvc/reference/RestTemplateCopyHeaderFilter.java

```diff
@@ -40,6 +40,10 @@ public void beforeSendRequest(Invocation invocation, HttpServletRequestEx reques
 
     httpHeaders.forEach((key, values) -> {
       for (String value : values) {
+        // null args should not be set to requestEx to avoid NullPointerException in Netty.
+        if (null == value) {
+          continue;
```

Comment:
Please do some debug log to help the user tracing the setting issue.

Reply:
debug log added, please review it.